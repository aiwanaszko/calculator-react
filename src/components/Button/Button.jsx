import React from "react";
import "./Button.scss";

const isOperator = (val) => {
  return val === "+" || val === "-" || val === "AC" || val === "C";
};

const isEqualityOperator = (val) => {
  return val === "=";
};

export default function Button(props) {
  return (
    <div
      className={`button-wrapper ${
        isOperator(props.children)
          ? "operator"
          : isEqualityOperator(props.children)
          ? "equality-operator"
          : ""
      }`}
      onClick={() => props.handleClick(props.children)}
    >
      {props.children}
    </div>
  );
}
