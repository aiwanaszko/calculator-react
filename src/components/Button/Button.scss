.button-wrapper {
  width: 90px;
  height: 90px;
  display: flex;
  justify-content: center;
  align-items: center;
  color: white;
  border-radius: 50%;
  background-color: #333333;
  font-size: 20px;
  margin-bottom: 10px;
  &:hover {
    cursor: pointer;
  }
}

.operator {
  background-color: #fe9f06;
  color: white;
}

.equality-operator {
  height: 180px;
  background-color: #fe9f06;
}
