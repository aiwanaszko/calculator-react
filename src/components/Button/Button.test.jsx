import React from "react";
import renderer from "react-test-renderer";
import Button from "./Button";

it("renders the Button correctly", () => {
  const component = renderer.create(<Button />).toJSON();
  expect(component).toMatchSnapshot();
});
