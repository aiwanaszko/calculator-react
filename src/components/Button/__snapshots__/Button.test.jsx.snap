// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Button correctly 1`] = `
<div
  className="button-wrapper "
  onClick={[Function]}
/>
`;
